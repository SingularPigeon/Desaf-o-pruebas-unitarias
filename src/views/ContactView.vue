<template>
  <h1>Contact us</h1>
  <form>
    <label for="name">Name:</label>
    <input type="text" id="name" name="name" v-model="name"><
    <label for="email">Email:</label> 
    <input type="email" id="email" name="email" v-model="email"><
    <label for="message">Message:</label>
    <textarea id="message" name="message" v-model="message"></textarea>
    <button @click.prevent="submitForm">Submit</button>
  </form>
</template>
<script setup>
import { ref } from 'vue';
const name = ref('');
const email = ref('');
const message = ref('');

</script>