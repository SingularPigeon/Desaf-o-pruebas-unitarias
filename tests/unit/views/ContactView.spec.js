import { mount} from '@vue/test-utils'

import ContactView from '@/views/ContactView.vue'
import { createRouter, createWebHistory } from 'vue-router'

describe('ContactView', () => {
  const wrapper = mount(ContactView)

  it('EL HTML mantiene su estructura', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

test('Comprobando la ruta a la vista ContactView', async () => {
  const router = createRouter({
    history: createWebHistory(),
    routes: [
      {
        path: '/contact',
        name: 'contact',
        component: ContactView
      }
    ]
})
router.push('/contact')
await router.isReady()
const wrapper = mount(ContactView, {
  global: {
    plugins: [router]
  }
})
expect(wrapper.findComponent(ContactView).exists()).toBe(true)
})

test('Tiene un formulario', async () => {
  const form = wrapper.find('form')
  expect(form.exists()).toBe(true)
 
})
})