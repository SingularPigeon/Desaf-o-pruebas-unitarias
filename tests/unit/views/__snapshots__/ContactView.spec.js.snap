// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactView EL HTML mantiene su estructura 1`] = `
<h1>Contact us</h1>
<form><label for="name">Name:</label><input type="text" id="name" name="name">&lt; <label for="email">Email:</label><input type="email" id="email" name="email">&lt; <label for="message">Message:</label><textarea id="message" name="message"></textarea><button>Submit</button></form>
`;
